salesloft/parse.go

```diff
@@ -0,0 +1,85 @@
+package salesloft
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/linkutils"
+	"github.com/spyzhov/ajson"
+)
+
+/*
+Response example:
+
+	{
+	  "metadata": {
+	    "filtering": {},
+	    "paging": {
+	      "per_page": 25,
+	      "current_page": 1,
+	      "next_page": 2,
+	      "prev_page": null
+	    },
+	    "sorting": {
+	      "sort_by": "updated_at",
+	      "sort_direction": "DESC NULLS LAST"
+	    }
+	  },
+	  "data": [...]
+	}
+*/
+func getTotalSize(node *ajson.Node) (int64, error) {
+	return common.JSONManager.ArrSize(node, "data")
+}
+
+func getRecords(node *ajson.Node) ([]map[string]any, error) {
+	arr, err := common.JSONManager.GetArr(node, "data")
+	if err != nil {
+		return nil, err
+	}
+
+	return common.JSONManager.ArrToMap(arr)
+}
+
+func makeNextRecordsURL(reqLink *linkutils.URL) common.NextPageFunc {
+	return func(node *ajson.Node) (string, error) {
+		nested, err := common.JSONManager.GetNestedNode(node, []string{"metadata", "paging"})
+		if err != nil {
+			if errors.Is(err, common.ErrKeyNotFound) {
+				// list resource doesn't support pagination, hence no next page
```

Comment:
What do you mean by "list resource doesn't support pagination"? Are you referring to a particular endpoint?

Reply:
Metadata sometimes doesn't have `paging` object. There are tests for these. Sample response that I am refering to is below.

**Reading users:**
https://github.com/amp-labs/connectors/blob/d6481dd91feba64de85faaab0105eeb917b3f564/salesloft/test/read-list-users.json#L2-L8
**Reading people:**
https://github.com/amp-labs/connectors/blob/d6481dd91feba64de85faaab0105eeb917b3f564/salesloft/test/read-list-people.json#L2-L14